topics/wallet-interaction-standard.md

```diff
@@ -0,0 +1,255 @@
+# ICRC-25: Wallet Interaction Standard
+
+## Summary
+
+This specification describes a communication protocol between dapps (decentralized applications) and wallets. It defines messages that both sides should use to interact with each other and provides guidelines on how to process them.
+
+## Terminology
+
+* wallet: A service that manages a user's keys and can sign and perform canister calls on their behalf.
+* relying party: A service that wants to request calls on a specific canister.
+
+## Messages
+
+### `permission`
+
+The purpose of the `permission` messages is to establish a connection between a relying party and a wallet, grant the relying party access to public parts of the user's identity and define the scope of actions the relying part is allowed to perform.
+
+#### Types
+
+- `text`: A plain `string` value.
+- `blob`: A `string` value describing binary data encoded in base64.
+
+#### Request
+
+`version` (`text`): The version of the standard used. If the wallet does not support the version of the request, it must send the `"VERSION_NOT_SUPPORTED"` error in response.
+
+`appMetadata`: Information about the relying party.
+- `name` (`text`): The user-friendly name of the relying party. 
+- `icon` (`text`, optional): An optional URL pointing to an icon resource representing the relying party.
+
+`networks`: A list of networks on which the relying party plans to operate.
+- `chainId`(`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes the relying party requires. Possible values: 
+- `"canister_call"`
+
+`challenge` (`blob`): A challenge used for the wallet to sign in order to prove its access to the identity. The challenge should be an array of 32 cryptographically random bytes generated from a secure random source by the sender of the request.
+
+#### Response
+
+`version` (`text`): The version of the standard used. It must match the `version` from the request.
```

Comment:
Version is defined as text here but in the example request it's a number? Looking at the JSON-RPC 2.0 spec, numbers are valid values besides text but things bigint will need to be encoded as string.

Reply:
I just went with the most versatile type here to open it to values other than just numbers, for example two- or three-part version numbers, or something more descriptive like, for example, `'1.icrcXX'` for extensions of this standard. However, I'm not sure, if there's a real use case for such versions 🤔 if we were to stick to just a single number versions, then I agree the type could be narrowed down to just a number.